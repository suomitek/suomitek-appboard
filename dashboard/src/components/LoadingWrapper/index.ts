import LoadingWrapper, { LoaderType } from "./LoadingWrapper";
export { LoaderType };
export default LoadingWrapper;
