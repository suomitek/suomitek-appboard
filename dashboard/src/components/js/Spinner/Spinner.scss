.spinner {
  &-container {
    &.spinner-center {
      display: block;
      text-align: center;
    }
  }

  &.spinner-inline {
    margin-right: 6px;
  }
}
