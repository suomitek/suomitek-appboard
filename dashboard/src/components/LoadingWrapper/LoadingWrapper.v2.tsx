import * as React from "react";

import Spinner from "../js/Spinner";

export interface ILoadingWrapperProps {
  loaded?: boolean;
  small?: boolean;
  medium?: boolean;
  children?: any;
}

function LoadingWrapper(props: ILoadingWrapperProps) {
  return props.loaded ? props.children : <Spinner medium={props.medium} small={props.small} />;
}

export default LoadingWrapper;
