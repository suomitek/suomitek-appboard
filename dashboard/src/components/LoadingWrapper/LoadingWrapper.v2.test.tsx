import { shallow } from "enzyme";
import * as React from "react";
import Spinner from "../js/Spinner";
import LoadingWrapper from "./LoadingWrapper.v2";

let props = {} as any;

const ChildrenComponent = () => <div>Hello dad!</div>;

const renderComponent = (p: any) => {
  return shallow(
    <LoadingWrapper {...p}>
      <ChildrenComponent />
    </LoadingWrapper>,
  );
};

describe("when loaded is false", () => {
  beforeEach(() => {
    props = {
      loaded: false,
    };
  });

  it("matches the snapshot", () => {
    const wrapper = renderComponent(props);
    expect(wrapper).toMatchSnapshot();
  });

  it("does not render any children", () => {
    const wrapper = renderComponent(props);
    expect(wrapper.find(ChildrenComponent)).not.toExist();
  });

  it("renders a Spinner", () => {
    const wrapper = renderComponent(props);
    expect(wrapper.find(Spinner)).toExist();
  });

  it("renders a mid size Spinner", () => {
    const wrapper = renderComponent({ ...props, medium: true });
    expect(wrapper.find(Spinner)).toExist();
    expect(wrapper.find(Spinner).prop("medium")).toBe(true);
  });

  it("renders a small Spinner", () => {
    const wrapper = renderComponent({ ...props, small: true });
    expect(wrapper.find(Spinner)).toExist();
    expect(wrapper.find(Spinner).prop("small")).toBe(true);
  });
});

describe("when loaded is true", () => {
  beforeEach(() => {
    props = {
      loaded: true,
    };
  });

  it("matches the snapshot", () => {
    const wrapper = renderComponent(props);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders it wrapped component", () => {
    const wrapper = renderComponent(props);
    expect(wrapper.find(ChildrenComponent)).toExist();
  });
});
