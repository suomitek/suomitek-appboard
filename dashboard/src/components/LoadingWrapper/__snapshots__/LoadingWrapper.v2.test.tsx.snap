// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loaded is false matches the snapshot 1`] = `
<Spinner
  center={true}
  inline={false}
  inverse={false}
  medium={false}
  small={false}
  text="Loading..."
/>
`;

exports[`when loaded is true matches the snapshot 1`] = `<ChildrenComponent />`;
